Document Azure service principal auth in snapshot/agent.mdx (#21942)
hashicorp/consul: website/content/commands/snapshot/agent.mdx

- `-azure-blob-container-name` - Container to use. Required for Azure blob storage, and setting this
  disables local storage.

- `-azure-blob-environment` - Environment to use. Defaults to AZUREPUBLICCLOUD. Other valid environments
  are AZURECHINACLOUD, AZUREGERMANCLOUD and AZUREUSGOVERNMENTCLOUD. Introduced in Consul 1.7.3.

> ~> These below options `azure-blob-service-principal-id`, `azure-blob-service-principal-secret`, and `azure-blob-tenant-id` are introduced in v1.20.1+ent.

Reviewer: ```suggestion
~> The following options were introduced in v1.20.1+ent.
```

> - `-azure-blob-service-principal-id` and `-azure-blob-service-principal-secret` - Use these parameters to authenticate using Service Principal (SPN) ID + Secret together.

Reviewer: ```suggestion
- `-azure-blob-service-principal-id` -
- `-azure-blob-service-principal-secret` -
```
These parameters should be listed and documented separately.

> - `-azure-blob-tenant-id` - Required when using Service Principal ID and Secret.

Reviewer: ```suggestion
- `-azure-blob-tenant-id` - The ID of the tenant that owns the Azure blob.
```
It should be reworded to something similar that clearly describes the value that is required by this field.

- `-azure-blob-container-name` - Container to use. Required for Azure blob storage, and setting this
  disables local storage.

- `-azure-blob-environment` - Environment to use. Defaults to AZUREPUBLICCLOUD. Other valid environments
  are AZURECHINACLOUD, AZUREGERMANCLOUD and AZUREUSGOVERNMENTCLOUD. Introduced in Consul 1.7.3.

~> The following options were introduced in v1.20.1+ent.

> - `-azure-blob-service-principal-id` -
> - `-azure-blob-service-principal-secret` -

Reviewer: These parameters are missing descriptions.

```suggestion
- `-azure-blob-service-principal-id` - The ID of the service principal that owns the blob object.
- `-azure-blob-service-principal-secret` - The secret of the service principal that owns the blob object.
```

The above text are _examples_ what should be here. Please refer to the Azure documentation to see how these parameters should be described.

> - `-azure-blob-tenant-id` - The ID of the tenant that owns the Azure blob.

Reviewer: Is this description correct per Azure's docs?

Author: Yes. This is tested in the internal lab.
